plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java

```diff
@@ -72,81 +77,46 @@
-    private MavenProject project;
+@Component
+    RepositorySystem repoSystem;
 
-    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
     private RepositorySystemSession session;
 
-    @Inject
-    private ArtifactResolver artifactResolver;
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    private List<RemoteRepository> repositories;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    MavenProject project;
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    MavenSession mavenSession;
 
     @Inject
     private CommandExecutor commandExecutor;
 
     /**
-     * The WildFly Application Server's home directory. If not used, WildFly will be downloaded.
+     * The WildFly Application Server's home directory. If not used, WildFly will be installed.
      */
     @Parameter(alias = "jboss-home", property = PropertyNames.JBOSS_HOME)
     private String jbossHome;
 
     /**
-     * A string of the form groupId:artifactId:version[:packaging][:classifier]. Any missing portion of the artifact
-     * will be replaced with the it's appropriate default property value
-     *
-     * @deprecated this will be removed in 3.0.0 in favor of provisioning a server
-     */
-    @Deprecated
-    @Parameter(property = PropertyNames.WILDFLY_ARTIFACT)
-    private String artifact;
-
-    /**
-     * The {@code groupId} of the artifact to download. Ignored if {@link #artifact} {@code groupId} portion is used.
-     *
-     * @deprecated this will be removed in 3.0.0 in favor of provisioning a server
-     */
-    @Deprecated
-    @Parameter(defaultValue = ArtifactNameBuilder.WILDFLY_GROUP_ID, property = PropertyNames.WILDFLY_GROUP_ID)
-    private String groupId;
-
-    /**
-     * The {@code artifactId} of the artifact to download. Ignored if {@link #artifact} {@code artifactId} portion is
-     * used.
-     *
-     * @deprecated this will be removed in 3.0.0 in favor of provisioning a server
-     */
-    @Deprecated
-    @Parameter(defaultValue = ArtifactNameBuilder.WILDFLY_ARTIFACT_ID, property = PropertyNames.WILDFLY_ARTIFACT_ID)
-    private String artifactId;
-
-    /**
-     * The {@code classifier} of the artifact to download. Ignored if {@link #artifact} {@code classifier} portion is
-     * used.
-     *
-     * @deprecated this will be removed in 3.0.0 in favor of provisioning a server
+     * The version of the WildFly default server to install in case no jboss-home has been set
+     * and no server has previously been provisioned.
+     * The latest stable version is resolved if left blank.
      */
-    @Deprecated
-    @Parameter(property = PropertyNames.WILDFLY_CLASSIFIER)
-    private String classifier;
+    @Parameter(alias = "version", property = PropertyNames.WILDFLY_VERSION)
```

Comment:
Not a huge deal, but `alias = "version"` shouldn't really be needed.

Reply:
Thanks, fixed.